### PR TITLE
VZ-3420. Move WKO install to platform operator.

### PR DESCRIPTION
**Pull request by michael-cico**
# Description

Move the WKO install to the platform operator.
- Added hooks for pre-install fn for a `helmComponent`, and added a WKO impl to set up the pre-reqs for that service
- Added the ability to declare dependencies between components in the registry, and validate that those dependencies are `Ready` before allowing install
- Declared dependency on Istio from WKO, and added Istio ready check fn
- Add the correct labels to the `verrazzano-system` namespace
- Added/updated the unit tests

Fixes VZ-3240.

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/registry.go [hunk 1]
```diff
@@ -148,3 +158,56 @@ func GetComponents() []Component {
...
+			return false
+		}
+	}
+	return true
+}
+
+// checkDependencies Check the ready state of any dependencies and check for cycles
+func checkDependencies(log *zap.SugaredLogger, client client.Client, c Component, trace map[string]bool) (map[string]bool, error) {
+	for _, dependencyName := range c.GetDependencies() {
+		if trace == nil {
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/registry.go:194]: do you plan on leaving the trace in?

File: platform-operator/controllers/verrazzano/controller.go [hunk 2]
```diff
@@ -785,15 +791,43 @@ func (r *Reconciler) createVerrazzanoSystemNamespace(ctx context.Context, log *z
...
+		return nil
+	}
+	if err := r.Update(ctx, &vzSystemNS); err != nil {
+		return err
+	}
+	return nil
+}
+
+// mergeMaps Merge one map into another, creating new one if necessary; returns the updated map and true if it was modified
+func mergeMaps(to map[string]string, from map[string]string) (map[string]string, bool) {
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/controller.go:818]: this should go in pkg.string maybe.  Not a blocker for merge.

File: platform-operator/internal/k8s/status/deployment_ready.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2021, Oracle and/or its affiliates.
```
Comment (pfmackin) [inline platform-operator/internal/k8s/status/deployment_ready.go:1]: I think this should go in k8s/deployment.  I was sort of confused when i saw status, thinking it was the VZ CR status.  Again, not a blocker, we will be doing some rearranging after you merge as discussed.

**Review by pfmackin — DISMISSED** [review]

**Comment by pfmackin** [comment]
As discussed, we need some follow up work on cleaning up the component struct.

File: platform-operator/controllers/verrazzano/component/registry.go (hunk 1 above)
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/registry.go:194]: The trace is to track cycles in the dependencies.

File: platform-operator/controllers/verrazzano/controller.go (hunk 2 above)
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/controller.go:818]: OK

File: platform-operator/internal/k8s/status/deployment_ready.go (hunk 3 above)
Comment (michael-cico) [reply platform-operator/internal/k8s/status/deployment_ready.go:1]: sounds good...wasn't quite sure where to put it.

**Review by pfmackin — APPROVED** [review]

File: platform-operator/internal/k8s/status/deployment_ready.go (hunk 3 above)
Comment (michael-cico) [reply platform-operator/internal/k8s/status/deployment_ready.go:1]: Tracking in VZ-3422

File: platform-operator/controllers/verrazzano/controller.go (hunk 2 above)
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/controller.go:818]: Tracking in VZ-3422

**Pull request merged**